From fc715a48bdaa80c6db500e086b453b8d6c86a28e Mon Sep 17 00:00:00 2001
From: Ambroz Bizjak <ambrop7@gmail.com>
Date: Sat, 19 Feb 2011 17:26:20 +0000
Subject: Don't decrease refcount of files returned by GFileIcon (bug #7309).

This fixes a crash when hovering CD drives with a special icon in
computer://.

Signed-off-by: Jannis Pohlmann <jannis@xfce.org>
---
diff --git a/NEWS b/NEWS
index 9b21960..6ccac5d 100644
--- a/NEWS
+++ b/NEWS
@@ -4,6 +4,8 @@
 - Fix properties dialog crash with files with unknown content type.
 - Fix crash when opening a mountable drive with no media (bug #7308).
   Reported and solved by Ambroz Bizjak.
+- Fix crash when hovering a CD drive with a special icon (bug #7309).
+  Reported and solved by Ambroz Bizjak.
 
 1.2.1
 =====
diff --git a/thunar/thunar-file.c b/thunar/thunar-file.c
index 1767f10..8caa422 100644
--- a/thunar/thunar-file.c
+++ b/thunar/thunar-file.c
@@ -2912,7 +2912,6 @@ thunar_file_get_icon_name (const ThunarFile   *file,
         {
           icon_file = g_file_icon_get_file (G_FILE_ICON (icon));
           icon_name = g_file_get_path (icon_file);
-          g_object_unref (icon_file);
         }
     }
   
--
cgit 
